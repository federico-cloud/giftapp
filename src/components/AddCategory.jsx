import React, { useState } from 'react';

export const AddCategory = ({setCategorias}) => {
  
    const [inputValue, setInputValue] = useState('Escriba aqui su categoria');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 0) {
            setCategorias( cats => [inputValue,...cats]);
            setInputValue('')
        }

    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}
