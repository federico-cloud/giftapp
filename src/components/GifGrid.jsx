import React, { useEffect, useState } from 'react'

export const GifGrid = ({categoria}) => {

    const [imagenes, setImagenes] = useState([]);

    useEffect( () => {
        getGifs();
    }, [])

    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=goku&limit=20&api_key=FQNhGq9VGKl9uLcBgROf6bKArM2jKtTR';
        const resp = await fetch(url);

        const {data} = await resp.json();

        const gifs = data.map( img => {
            return{
                id: img.id,
                title: img.title,
                url: img.images.downsized_medium.url
            }
        })
        setImagenes(gifs);
    } 
  
    return (
        <>
            <h3>{categoria}</h3>
            <ol>
                {
                    imagenes.map( img => (
                        <li key={img.id}> {img.title}</li>
                    ))
                }
            </ol>
        </>
    )
}
