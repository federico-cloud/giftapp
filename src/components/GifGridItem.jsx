import React from 'react';
import PropTypes from 'prop-types';


export const GifGridItem = ( { title ,url } ) => {

    return (
        <div className="card animate__animated animate__zoomIn">
            <p>{title}</p>
            <img src={url} alt={title}/>
        </div>
    )

}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}