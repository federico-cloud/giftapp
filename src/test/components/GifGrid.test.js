import React from 'react';
import {shallow} from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en GifGrid', () => {

    const categoria = 'Goku';

    test('Debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid categoria={categoria} />)
        
        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostrar imagenes cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'http://example.com',
            title: 'Cualquier imagen'
        },{
            id: 123,
            url: 'http://example.com/gif',
            title: 'Otra imagen'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid categoria={categoria} />)
      
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })

})