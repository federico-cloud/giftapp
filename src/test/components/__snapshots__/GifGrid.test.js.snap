// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrar imagenes cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="container"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier imagen"
      url="http://example.com"
    />
    <GifGridItem
      id={123}
      key="123"
      title="Otra imagen"
      url="http://example.com/gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <h2>
    Loading
  </h2>
  <div
    className="container"
  />
</Fragment>
`;
